source/tyr/tyr/resources.py

```diff
@@ -657,6 +657,9 @@ def put(self, user_id):
         if not billing_plan:
             return ({'error': 'billing_plan doesn\'t exist'}, 400)
 
+        if args['shape'] == {}:
+            args['shape'] = ujson.loads(user.shape)
```

Comment:
No, that's the tricky special case of the empty object saying "no modification", thus you don't have default value as you provided the empty object